### ISSUE TITLE
CuratorZookeeperRepository.get might return old value, it make scaling job could not be created

### ISSUE THREAD

**Issue by sandynz** [apache/shardingsphere#15959]
### Which version of ShardingSphere did you use?
master branch, commit e5cb7628ebcda4b5c78cadcf97be61c8ccc3c407

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy, scaling enabled

### Expected behavior
Scaling job start successfully.

### Actual behavior
Scaling job could not start.
Error log:
```
[INFO ] 2022-03-10 11:17:29.041 [ShardingSphere-Command-2] o.a.s.m.m.c.c.r.c.s.ScalingRegistrySubscriber - start scaling job, locked the schema name, event=org.apache.shardingsphere.mode.manager.cluster.coordinator.registry.config.event.version.SchemaVersionPreparedEvent@419dc7d9
[INFO ] 2022-03-10 11:17:29.244 [ShardingSphere-Command-2] o.a.s.d.p.s.r.RuleAlteredJobWorker - Start scaling job by StartScalingEvent{schemaName='scaling_bmsql', schemaVersion='1'}
[INFO ] 2022-03-10 11:17:29.277 [ShardingSphere-Command-2] o.a.s.d.p.s.r.RuleAlteredJobWorker - type=org.apache.shardingsphere.sharding.yaml.config.YamlShardingRuleConfiguration, ruleAltered=false
[ERROR] 2022-03-10 11:17:29.278 [ShardingSphere-Command-2] o.a.s.d.p.s.r.RuleAlteredJobWorker - no altered rule
三月 10, 2022 11:17:29 上午 com.google.common.eventbus.EventBus$LoggingHandler handleException
严重: Exception thrown by subscriber method start(org.apache.shardingsphere.mode.manager.cluster.coordinator.registry.cache.event.StartScalingEvent) on subscriber org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJobWorker@709d6de5 when dispatching event: StartScalingEvent{schemaName='scaling_bmsql', schemaVersion='1'}
org.apache.shardingsphere.data.pipeline.core.exception.PipelineJobCreationException: no altered rule
	at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJobWorker.createJobConfig(RuleAlteredJobWorker.java:208)
	at org.apache.shardingsphere.data.pipeline.scenario.rulealtered.RuleAlteredJobWorker.start(RuleAlteredJobWorker.java:180)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:87)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:144)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:72)
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:67)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:108)
	at com.google.common.eventbus.EventBus.post(EventBus.java:212)
	at org.apache.shardingsphere.proxy.backend.text.distsql.rdl.rule.RuleDefinitionBackendHandler.prepareScaling(RuleDefinitionBackendHandler.java:160)
	at org.apache.shardingsphere.proxy.backend.text.distsql.rdl.rule.RuleDefinitionBackendHandler.execute(RuleDefinitionBackendHandler.java:91)
	at org.apache.shardingsphere.proxy.backend.text.distsql.rdl.rule.RuleDefinitionBackendHandler.execute(RuleDefinitionBackendHandler.java:60)
	at org.apache.shardingsphere.proxy.backend.text.SchemaRequiredBackendHandler.execute(SchemaRequiredBackendHandler.java:51)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.execute(MySQLComQueryPacketExecutor.java:67)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:97)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:69)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

### Reason analyze (If you can)
Summary: `CuratorZookeeperRepository.get` might return old value. When it return old rule, then there's no altered sharding rule, so no scaling job could be created.

Details:
After I ran `ALTER SHARDING TABLE RULE` (change some tables' `sharding-count` from `2` to `6`), there's a new version `1` with new rule. Screenshot:
<img width="831" alt="WX20220310-114610@2x" src="https://user-images.githubusercontent.com/42492540/157585937-98c56f02-d134-4890-a920-62213c646bb0.png">
When I re-ran `ALTER SHARDING TABLE RULE`, it still doesn't work. Some screenshots with debugging:
The path `/metadata/scaling_bmsql/versions/1/rules` is correct used for `CuratorZookeeperRepository.get`:
<img width="1385" alt="WX20220310-114837@2x" src="https://user-images.githubusercontent.com/42492540/157586078-d260e038-9d0a-4b35-a887-256b8751b142.png">
But the value of `/metadata/scaling_bmsql/versions/1/rules` is wrong, it's the content of version `0`:
<img width="1404" alt="WX20220310-115111@2x" src="https://user-images.githubusercontent.com/42492540/157586223-0c8e3eec-14e9-4935-9b07-91a416a0de40.png">
After several minutes, when I ran `repository.get(SchemaMetaDataNode.getRulePath(event.getSchemaName(), event.getVersion()))`, it got correct value of version `1`:
<img width="1407" alt="WX20220310-115257@2x" src="https://user-images.githubusercontent.com/42492540/157586317-a413c356-68f8-466e-bfb5-41281fbd8697.png">

Seems it's similar issue as #15752, but it affects scaling main process this time.

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
It could not be reproducued every time, once it occur, do not restart proxy, rule the same `ALTER SHARDING TABLE RULE` DistSQL to emit scaling job, then we could reproduct it, verify by debugging.

**Comment by menghaoranss** [comment]
@sandynz I will follow up.

**Comment by menghaoranss** [comment]
fixes by #15986
